Issue #1

rezil opened the pull request:
Issue #1 implemented as described. Horizontal scrolling header view with tabs that stick once the user scrolls all the way to the top. List content changes depending on the selected tab.

klemzy commented:
Thanks for the sent solution. I have a few comments. Current solution does not compile. WeatherApi and Weather clesses are missing. Can you push the missing classes?
I have noticed a few issues with the issue #1 implementation. Maybe the video in the issue #1 does not show the expected beviour very well. 
When you scroll the list it should scroll over the cat view and the cat view should stay in place. 
When scrolling the list it should move over the cat view until it reaches the top and then continue scrolling. Right now if you fling it already scrolls even though it hasn't reached the top.  
Also the list should scroll freely without any snapping, so if list is scrolled all the way to the bottom, if you fling it, it should scroll until reaching the top and then eventually show the cat view.
 Hopefully I was clear enough. If you have any questions don't hesitate to ask.

rezil commented:
Hello, apologies for the missing classes. They've been added now. Should swiping left and right on the list below the cat view and the tabs also change the content of the list? In the example movie the only way that the list content changed was by clicking on the tabs above the list itself.

klemzy commented:
Great. No need for the swiping on the list just the cat view. Changing content by clicking on the tab will suffice.

rezil commented:
I see what you mean by keeping the cat view stationary. I have misinterpreted the behavior described in issue #1 and am working on fixing it right now. 

About issue #2, could you go into a bit more detail as to how you expect error handling, the location request strategy etc. to work. For example, in the current implementation of #2 a single location is retrieved, then the location updates stop until a configuration change happens or the user reenters the app. Is it okay if it works like that or should I implement a different strategy? Similarly for errors, currently there's not much error handling - the app doesn't crash but the user does not get any feedback about it. The issue description does not say anything about error handling at all so I left it out.

klemzy commented:
Ok great. No worries about that. Your current implementation will suffice for the scope of this assignment.

rezil commented:
So, currently the app works as described in issue #1, I fixed the layout and added some quick and dirty code so it resembles #1. I have to clean the code up a bit and will probably submit the commit for issue #2 in the next hour or so.

rezil commented:
Code cleaned up and issue #2 implemented with the new layout. Let me know if I missed anything else in the requirements.
